docs/pages/_document.js

```diff
@@ -92,55 +93,70 @@ export default class MyDocument extends Document {
   }
 }
 
+// `getInitialProps` belongs to `_document` (instead of `_app`),
+// it's compatible with static-site generation (SSG).
 MyDocument.getInitialProps = async (ctx) => {
   // Resolution order
   //
   // On the server:
-  // 1. page.getInitialProps
-  // 2. document.getInitialProps
-  // 3. page.render
-  // 4. document.render
+  // 1. app.getInitialProps
+  // 2. page.getInitialProps
+  // 3. document.getInitialProps
+  // 4. app.render
+  // 5. page.render
+  // 6. document.render
   //
   // On the server with error:
-  // 2. document.getInitialProps
+  // 1. document.getInitialProps
+  // 2. app.render
   // 3. page.render
   // 4. document.render
   //
   // On the client
-  // 1. page.getInitialProps
-  // 3. page.render
+  // 1. app.getInitialProps
+  // 2. page.getInitialProps
+  // 3. app.render
+  // 4. page.render
 
   // Render app and page and get the context of the page with collected side effects.
-  const sheets = new ServerStyleSheets();
+  const materialSheets = new ServerStyleSheets();
+  const styledComponentsSheet = new ServerStyleSheet();
   const originalRenderPage = ctx.renderPage;
 
-  ctx.renderPage = () =>
-    originalRenderPage({
-      enhanceApp: (App) => (props) => sheets.collect(<App {...props} />),
-    });
+  try {
```

Comment:
Support for `styled-components` for next.js. There is still the warning: `Warning: Prop className did not match.`, but the page work as expected..

Reply:
> but the page work as expected..

Careful with these statements especially for hydration. Even if you did take a screenshot before and after hydration there could still be styles that mismatch that only apply to UI revealed after interaction. 

So this needs to be fixed before a stable release. Every warning we have that isn't actionable generates noise which is confusing to beginners and erodes trust in these warnings.

Reply:
Should we configure emotion too before moving forward? I guess we could also have it as a follow up task.

Reply:
Will work tomorrow on both configuring emption and solving the styled components warning

Reply:
Seems like emotion above 10 doesn't need any config for next.js🥳: https://emotion.sh/docs/ssr#nextjs
![image](https://user-images.githubusercontent.com/4512430/93561401-655fd500-f984-11ea-9434-c914fd23fdb7.png)

Reply:
Regarding the warning with styled-components, it still shows after you do the changes and refresh the page. I have tried everything I found on the matter and stamble apon this comment in the end - https://github.com/vercel/next.js/issues/7322#issuecomment-502465964

![image](https://user-images.githubusercontent.com/4512430/93660389-9c022200-fa4e-11ea-9b60-0faffc5b0f12.png)

So I will ignore it for now, as it is anyway not our default build, it's just for local testing of styled-components.